statics/js/components/preference.js

```diff
@@ -0,0 +1,142 @@
+/* jshint multistr: true */
+
+var PreferenceComponent = {
+
+  name: 'Preference',
+
+  mixins: [apiMixin, notificationMixin],
+
+  template: '\
+    <form class="form-preference" @submit.prevent="save">\
+      <div class="form-group">\
+        <label for="default-filter">Filters</label>\
+        <a><i class="fa fa-question help-text" aria-hidden="true" title="filter and highlight gremlin queries, displayed in the left panel"></i></a>\
+        <div v-for="filter in filters">\
+          <div class="form-group">\
+            <div class="input-group">\
+              <label class="input-group-addon">Name: </label>\
+              <input class="form-control" v-model="filter.name"/>\
+            </div>\
+            <div class="input-group filter-field">\
+              <label class="input-group-addon">Filter:  </label>\
+              <input class="form-control" v-model="filter.filter"/>\
+            </div>\
+          </div>\
+        </div>\
+        <button class="btn btn-primary btn-round-xs btn-xs" type="button" @click="addFilter">+</button>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-threshold">Bandwidth Threshold</label>\
+        <a><i class="fa fa-question help-text" aria-hidden="true" title="bandwidth threshold mode (relative/absolute)"></i></a>\
+        <select id="be-threshold" v-model="bwThreshold" class="form-control input-sm">\
+          <option value="absolute">Absolute</option>\
+          <option value="relative">Relative</option>\
+        </select>\
+      </div>\
+      <div v-if="bwThreshold == \'absolute\'">\
+        <div class="form-group">\
+          <label for="bw-abs-active">Bandwidth Absolute Active</label>\
+          <a><i class="fa fa-question help-text" aria-hidden="true" title="Active threshold in kbps"></i></a>\
+          <input id="bw-abs-active" type="number" class="form-control input-sm" v-model="bwAbsActive" min="0"/>\
+        </div>\
+        <div class="form-group">\
+          <label for="bw-abs-warning">Bandwidth Absolute Warning</label>\
+          <a><i class="fa fa-question help-text" aria-hidden="true" title="Warning threshold in kbps"></i></a>\
+          <input id="bw-abs-warning" type="number" class="form-control input-sm" v-model="bwAbsWarning" min="0"/>\
+        </div>\
+        <div class="form-group">\
+          <label for="bw-abs-alert">Bandwidth Absolute Alert</label>\
+          <a><i class="fa fa-question help-text" aria-hidden="true" title="Alert threshold in kbps"></i></a>\
+          <input id="bw-abs-alert" type="number" class="form-control input-sm" v-model="bwAbsAlert" min="0"/>\
+        </div>\
+      </div>\
+      <div v-if="bwThreshold == \'relative\'">\
+        <div class="form-group">\
+          <label for="bw-rel-active">Bandwidth Relative Active</label>\
```

Comment:
could you add a help note for these fields too saying that the value is a ratio between 0-1 or something similar

Reply:
done